api/client/ffs.go

```diff
@@ -236,6 +273,54 @@ func (f *ffs) Get(ctx context.Context, c cid.Cid) (io.Reader, error) {
 	return reader, nil
 }
 
+func (f *ffs) WatchLogs(ctx context.Context, ch chan<- LogEvent, c cid.Cid, opts ...WatchLogsOption) (func(), error) {
```

Comment:
Hey @jsign I just added this client method for `WatchLogs`. I'm trying the same style as the API implementation on the server side where the user passes in the `chan`. Not that the user is passing in the `chan` do you see any way we can simplify the cancelation part of this, or is it still necessary to return a cancel func (or tell the user to cancel the context they passed in)?

(I was thinking maybe something like they could just close the `chan` they passed in, but I don't know if that is a legit pattern in Go)

Reply:
Yes, the `cancel()` function can be avoided, and I think it _should_ be avoided.
The proper way to signal that the client is done is by `ctx` cancellation. 
Closing the channel would make the other part panic, since sending to a closed channel causes a panic.

That context cancellation is what's usually more idiomatic in Go, and it's what I used in on the server side.

Reply:
Ok, got rid of those cancel functions. Made a little change to deal with the error that gets propagated to the client when it cancels the context. Will comment in the code and would like you feedback there.